command-expansion-server/src/app.ts

```diff
@@ -118,10 +123,51 @@ app.post('/put-expansion', async (req, res, next) => {
 });
 
 app.post('/put-expansion-set', async (req, res, next) => {
+  const context: Context = res.locals.context;
+
   const commandDictionaryId = req.body.input.commandDictionaryId as number;
   const missionModelId = req.body.input.missionModelId as number;
   const expansionIds = req.body.input.expansionIds as number[];
 
+  const [expansions, commandTypes] = await Promise.all([
+    context.expansionDataLoader.loadMany(expansionIds.map(id => ({ expansionId: id }))),
+    context.commandTypescriptDataLoader.load({ dictionaryId: commandDictionaryId }),
+  ]);
+
+  const typecheckErrorPromises = await Promise.allSettled(expansions.map(async (expansion, index) => {
+    if (expansion instanceof Error) {
+      throw new InheritedError(`Expansion with id: ${expansionIds[index]} could not be loaded`, expansion);
+    }
+    const activitySchema = await context.activitySchemaDataLoader.load({missionModelId, activityTypeName: expansion.activityType });
+    const activityTypescript = generateTypescriptForGraphQLActivitySchema(activitySchema);
+    const result = await (piscina.run({
+      expansionLogic: expansion.expansionLogic,
+      commandTypes: commandTypes,
+      activityTypes: activityTypescript,
+    }, { name: 'typecheckExpansion' }) as ReturnType<typeof typecheckExpansion>);
+
+    return result.errors;
+  }));
+
+  const errors = unwrapPromiseSettledResults(typecheckErrorPromises).reduce((accum, item) => {
+    if (item instanceof Error) {
+      accum.push(item);
+    } else {
+      accum.concat(item);
+    }
```

Comment:
It's an error if it: fails to load the activity schema or there is some kind of panic-style failure. Otherwise, it is an array of diagnostic style errors. I think it would be good for use to pick and use a more specific/different nomenclature to differentiate between panic errors and diagnostic errors (user errors).